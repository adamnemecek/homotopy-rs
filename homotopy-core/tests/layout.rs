use homotopy_core::{examples, layout::Layout};
use insta::assert_debug_snapshot;

#[test]
fn monoid_unit() {
    let (_, diagram) = examples::monoid_unit();
    assert_debug_snapshot!(Layout::<2>::new(&diagram.into()).unwrap());
}

#[test]
fn scalar_and_beads() {
    let (_, diagram) = examples::scalar_and_beads();
    assert_debug_snapshot!(Layout::<2>::new(&diagram.into()).unwrap());
}

#[test]
fn associator() {
    let (_, diagram) = examples::associator();
    assert_debug_snapshot!(Layout::<3>::new(&diagram.into()).unwrap());
}

#[test]
fn snake() {
    let (_, diagram) = examples::algebraic_snake();
    assert_debug_snapshot!(Layout::<3>::new(&diagram.into()).unwrap());
}

#[test]
fn lips() {
    let (_, diagram) = examples::lips();
    assert_debug_snapshot!(Layout::<4>::new(&diagram.into()).unwrap());
}

#[test]
fn pants_unit() {
    let (_, diagram) = examples::pants_unit();
    assert_debug_snapshot!(Layout::<4>::new(&diagram.into()).unwrap());
}
