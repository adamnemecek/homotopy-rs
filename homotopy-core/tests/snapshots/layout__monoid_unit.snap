---
source: homotopy-core/tests/layout.rs
expression: layout
---
Layout(
    {
        [
            Boundary(
                Target,
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            8.0,
        ],
        [
            Boundary(
                Target,
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            2.0,
            8.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.0,
            1.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Interior(
                Singular(
                    1,
                ),
            ),
        ]: [
            5.0,
            4.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            1.0,
        ],
        [
            Interior(
                Regular(
                    1,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            3.0,
        ],
        [
            Interior(
                Singular(
                    2,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            6.0,
        ],
        [
            Boundary(
                Source,
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Interior(
                Regular(
                    2,
                ),
            ),
        ]: [
            6.0,
            5.0,
        ],
        [
            Boundary(
                Target,
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            4.0,
            8.0,
        ],
        [
            Interior(
                Singular(
                    0,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            2.0,
        ],
        [
            Interior(
                Regular(
                    1,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            3.0,
            3.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            3.0,
            5.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            5.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            2.0,
            1.0,
        ],
        [
            Boundary(
                Target,
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            5.5,
            8.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            4.0,
            4.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Interior(
                Regular(
                    2,
                ),
            ),
        ]: [
            6.0,
            4.0,
        ],
        [
            Boundary(
                Source,
            ),
            Interior(
                Singular(
                    1,
                ),
            ),
        ]: [
            4.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Interior(
                Regular(
                    2,
                ),
            ),
        ]: [
            5.5,
            1.0,
        ],
        [
            Interior(
                Singular(
                    0,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            3.0,
            2.0,
        ],
        [
            Interior(
                Regular(
                    1,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            3.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            4.0,
        ],
        [
            Boundary(
                Source,
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    1,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.5,
            3.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.5,
            5.0,
        ],
        [
            Interior(
                Regular(
                    3,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            7.0,
        ],
        [
            Boundary(
                Source,
            ),
            Interior(
                Regular(
                    2,
                ),
            ),
        ]: [
            5.5,
            0.0,
        ],
        [
            Boundary(
                Source,
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            3.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Interior(
                Singular(
                    1,
                ),
            ),
        ]: [
            5.0,
            5.0,
        ],
        [
            Interior(
                Regular(
                    3,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            2.0,
            7.0,
        ],
        [
            Boundary(
                Target,
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            8.0,
        ],
        [
            Interior(
                Singular(
                    2,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            6.0,
            6.0,
        ],
        [
            Interior(
                Singular(
                    0,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.0,
            2.0,
        ],
        [
            Boundary(
                Source,
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            2.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            3.0,
            1.0,
        ],
        [
            Interior(
                Singular(
                    2,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            4.0,
            6.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            3.0,
            4.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            4.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            1.0,
        ],
        [
            Interior(
                Singular(
                    2,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            6.0,
        ],
        [
            Interior(
                Regular(
                    3,
                ),
            ),
            Interior(
                Singular(
                    0,
                ),
            ),
        ]: [
            4.0,
            7.0,
        ],
        [
            Interior(
                Regular(
                    1,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            5.0,
            3.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            4.0,
            5.0,
        ],
        [
            Interior(
                Regular(
                    3,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            5.5,
            7.0,
        ],
        [
            Interior(
                Singular(
                    0,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            2.0,
        ],
        [
            Boundary(
                Source,
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.0,
            0.0,
        ],
        [
            Interior(
                Regular(
                    2,
                ),
            ),
            Boundary(
                Source,
            ),
        ]: [
            0.0,
            5.0,
        ],
        [
            Interior(
                Regular(
                    3,
                ),
            ),
            Boundary(
                Target,
            ),
        ]: [
            7.0,
            7.0,
        ],
        [
            Interior(
                Regular(
                    0,
                ),
            ),
            Interior(
                Singular(
                    1,
                ),
            ),
        ]: [
            4.0,
            1.0,
        ],
        [
            Interior(
                Singular(
                    2,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.5,
            6.0,
        ],
        [
            Interior(
                Singular(
                    1,
                ),
            ),
            Interior(
                Regular(
                    0,
                ),
            ),
        ]: [
            1.5,
            4.0,
        ],
        [
            Interior(
                Singular(
                    0,
                ),
            ),
            Interior(
                Regular(
                    1,
                ),
            ),
        ]: [
            5.5,
            2.0,
        ],
    },
)
